src/System.Buffers.Primitives/System/Internal/OwnedArray.cs

```diff
@@ -1,63 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System.Runtime;
-
-namespace System.Buffers.Internal
-{
-    internal sealed class OwnedArray<T> : ReferenceCountedBuffer<T>
-    {
-        T[] _array;
-
-        public OwnedArray(int length)
-        {
-            _array = new T[length];
-        }
-
-        public OwnedArray(T[] array)
-        {
-            _array = array;
-        }
-
-        public OwnedArray(ArraySegment<T> segment)
```

Comment:
What is the motivation to remove the OwnedArray constructor that takes an ArraySegment?